webtau-core/src/main/java/com/twosigma/webtau/Matchers.java

```diff
@@ -0,0 +1,268 @@
+     * Contain matcher
+     * <pre>
+     * actual(value).should(contain(10));
+     * </pre>
+     * @param expected value to be contained
+     * @return matcher instance
+     */
+    public static ContainMatcher contain(Object expected) {
+        return new ContainMatcher(expected);
+    }
+
+    /**
+     * Contain matcher. Alias to contain
+     * <pre>
+     * actual(value).should(containing(10));
+     * </pre>
+     * @param expected value to be contained
+     * @return matcher instance
+     */
+    public static ContainMatcher containing(Object expected) {
+        return new ContainMatcher(expected);
+    }
+
+    /**
+     * Greater than matcher
+     * <pre>
+     * actual(value).shouldBe(greaterThan(10));
+     * </pre>
+     * @param expected value to be greater than
+     * @return matcher instance
+     */
+    public static GreaterThanMatcher greaterThan(Object expected) {
+        return new GreaterThanMatcher(expected);
+    }
+
+    /**
+     * Greater than or equal matcher
+     * <pre>
+     * actual(value).shouldBe(greaterThanOrEqual(10));
+     * </pre>
+     * @param expected value to be greater than or equal
+     * @return matcher instance
+     */
+    public static GreaterThanOrEqualMatcher greaterThanOrEqual(Object expected) {
+        return new GreaterThanOrEqualMatcher(expected);
+    }
+
+    /**
+     * Less than matcher
+     * <pre>
+     * actual(value).shouldBe(lessThan(10));
+     * </pre>
+     * @param expected value to be less than
+     * @return matcher instance
+     */
+    public static LessThanMatcher lessThan(Object expected) {
+        return new LessThanMatcher(expected);
+    }
+
+    /**
+     * Less than or equal matcher
+     * <pre>
+     * actual(value).shouldBe(lessThanOrEqual(10));
+     * </pre>
+     * @param expected value to be less than
+     * @return matcher instance
+     */
+    public static LessThanOrEqualMatcher lessThanOrEqual(Object expected) {
+        return new LessThanOrEqualMatcher(expected);
+    }
+
+    /**
+     * Throw exception <code>code</code> matcher.
+     * <pre>
+     * code(() -> {
+     *     businessLogic(-10);
+     * }).should(throwException("negative are not allowed"));
+     * </pre>
+     * @see #code(CodeBlock)
+     *
+     * @param expectedMessage expected exception message
+     * @return matcher instance
+     */
+    public static ThrowExceptionMatcher throwException(String expectedMessage) {
+        return new ThrowExceptionMatcher(expectedMessage);
+    }
+
+    /**
+     * Throw exception <code>code</code> matcher.
+     * <pre>
+     * code(() -> {
+     *      businessLogic(-10);
+     * }).should(throwException(Pattern.compile("negative .* not allowed")));
+     * </pre>
+     * @see #code(CodeBlock)
+     *
+     * @param expectedMessageRegexp regular pattern to match expected exception message
+     * @return matcher instance
+     */
+    public static ThrowExceptionMatcher throwException(Pattern expectedMessageRegexp) {
```

Comment:
Should we add an alias to this one which takes a string and is called something like `throwExceptionWithMessageMatching`?

Reply:
Have mixed feelings about this one. May start going into the direction of multiple overloads like junit does: `assertEquals`, `assertEqualsArrays`, `assertEqualsNumbers`. 
It is readable on the client side, at least Groovy portion: 
```
throwException(~/negative/)
```
But as you pointed before, it could be just because I know how it works.

Reply:
I'm happy to leave it out until someone asks for it.

Reply:
I think the overlo[ar]ds are sufficiently expressive, but then again I only use this with groovy.